src/chain/mod.rs

```diff
@@ -578,16 +596,19 @@ impl Filter for ChainSource {
 
 fn periodically_archive_fully_resolved_monitors(
 	channel_manager: Arc<ChannelManager>, chain_monitor: Arc<ChainMonitor>,
-	latest_channel_monitor_archival_height: Arc<RwLock<Option<u32>>>,
-) {
-	let mut latest_archival_height_lock = latest_channel_monitor_archival_height.write().unwrap();
+	kv_store: Arc<DynStore>, logger: Arc<FilesystemLogger>, node_metrics: Arc<RwLock<NodeMetrics>>,
+) -> Result<(), Error> {
+	let mut locked_node_metrics = node_metrics.write().unwrap();
 	let cur_height = channel_manager.current_best_block().height;
-	let should_archive = latest_archival_height_lock
+	let should_archive = locked_node_metrics
+		.latest_channel_monitor_archival_height
 		.as_ref()
 		.map_or(true, |h| cur_height >= h + RESOLVED_CHANNEL_MONITOR_ARCHIVAL_INTERVAL);
```

Comment:
I guess unrelated to this PR, but it's unclear to me why we don't archive on every block. IIUC, this saying to do so every six blocks.